stages/org.osbuild.grub2

```diff
@@ -38,8 +42,62 @@ and accompanying data can be installed from the built root via `uefi.install`.
 Both UEFI and Legacy can be specified at the same time.
 """
 STAGE_OPTS = """
-"required": ["root_fs_uuid"],
+"oneOf": [{
+  "required": ["root_fs_uuid"]
+}, {
+  "required": ["rootfs"]
+}],
 "properties": {
+  "rootfs": {
+    "description": "How to locate the root file system",
+    "type": "object",
+    "oneOf": [{
+      "required": ["uuid"]
+    }, {
+      "required": ["label"]
+    }],
+    "properties": {
+      "label": {
+        "description": "Identify the file system by label",
+        "type": "string"
+      },
+      "uuid": {
+        "description": "Identify the file system by UUID",
+        "type": "string",
+        "oneOf": [
+          { "pattern": "^[0-9A-Za-z]{8}(-[0-9A-Za-z]{4}){3}-[0-9A-Za-z]{12}$",
+            "examples": ["9c6ae55b-cf88-45b8-84e8-64990759f39d"] },
+          { "pattern": "^[0-9A-Za-z]{4}-[0-9A-Za-z]{4}$",
+            "examples": ["6699-AFB5"] }
+        ]
+      }
+    }
+  },
+  "bootfs": {
```

Comment:
Is there a way to refer to a type that's shared between `bootfs` and `rootfs`?

Reply:
I was wondering the same. Let me dig into the spec.

Reply:
Aha, [`definitions`](https://json-schema.org/understanding-json-schema/structuring.html#reuse)

Reply:
Done.

Reply:
Lovely. Thanks :heart: